<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from /home/hzx/hzx/leapting_code/model_and_simulator/lt_description/urdf/inc/test_kr1100.urdf.xacro | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="kr1100">
    <material name="Blue">
        <color rgba="0 0 1 1" />
    </material>
    <link name="base_arm">
        <inertial>
            <origin rpy="0 0 0" xyz="-0.046785 -0.00098204 0.10974" />
            <mass value="100.64" />
            <!-- value="91.64" /> -->
            <inertia ixx="0.60334" ixy="-0.00040839" ixz="-0.0038944" iyy="0.68726" iyz="0.00020595" izz="0.5917" />
        </inertial>
        <visual>
            <origin rpy="0 0 0" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://lt_description/meshes/kr10r1100/base_arm.STL" />
            </geometry>
            <material name="">
                <color rgba="0.4 0.4 0.4 1" />
            </material>
        </visual>
        <collision>
            <origin rpy="0 0 0" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://lt_description/meshes/kr10r1100/base_arm.STL" />
            </geometry>
        </collision>
    </link>
    <link name="link_1">
        <inertial>
            <origin rpy="0 0 0" xyz="0.0080148 -0.0054241 0.11231" />
            <mass value="0.01" />
            <!-- value="45.516" /> -->
            <inertia ixx="0.52733" ixy="-0.034725" ixz="-0.03364" iyy="0.35598" iyz="0.020992" izz="0.40114" />
        </inertial>
        <visual>
            <origin rpy="0 0 0" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://lt_description/meshes/kr10r1100/link_1.STL" />
            </geometry>
            <material name="">
                <color rgba="0.8 0.8 0.8 1" />
            </material>
        </visual>
        <collision>
            <origin rpy="0 0 0" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://lt_description/meshes/kr10r1100/link_1.STL" />
            </geometry>
        </collision>
    </link>
    <joint name="joint_a1" type="revolute">
        <origin rpy="0 0 0" xyz="-0.00061909 0 0.2157" />
        <parent link="base_arm" />
        <child link="link_1" />
        <axis xyz="0 0 -1" />
        <limit effort="100" lower="-3.220132469929538" upper="3.2288591161895095" velocity="1" />
    </joint>
    <link name="link_2">
        <inertial>
            <origin rpy="0 0 0" xyz="0.26612 -0.02538 0.068776" />
            <mass value="0.01" />
            <!-- value="84.384" /> -->
            <inertia ixx="0.443" ixy="0.22357" ixz="-0.027695" iyy="3.8805" iyz="0.00077028" izz="3.6823" />
        </inertial>
        <visual>
            <origin rpy="0 0 0" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://lt_description/meshes/kr10r1100/link_2.STL" />
            </geometry>
            <material name="">
                <color rgba="0.8 0.8 0.8 1" />
            </material>
        </visual>
        <collision>
            <origin rpy="0 0 0" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://lt_description/meshes/kr10r1100/link_2.STL" />
            </geometry>
        </collision>
    </link>
    <joint name="joint_a2" type="revolute">
        <origin rpy="-1.7871 -1.5708 0" xyz="0.011395 -0.071615 0.1803" />
        <parent link="link_1" />
        <child link="link_2" />
        <axis xyz="0 0 1" />
        <limit effort="100" lower="-2.443460952792061" upper="2.4598670477608082" velocity="1" />
    </joint>
    <link name="link_3">
        <inertial>
            <origin rpy="0 0 0" xyz="0.011131 0.052479 0.072757" />
            <mass value="0.01" />
            <!-- value="31.013" /> -->
            <inertia ixx="0.21568" ixy="-0.02434" ixz="-0.00045379" iyy="0.080696" iyz="-0.0033315" izz="0.22357" />
        </inertial>
        <visual>
            <origin rpy="0 0 0" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://lt_description/meshes/kr10r1100/link_3.STL" />
            </geometry>
            <material name="">
                <color rgba="0.8 0.8 0.8 1" />
            </material>
        </visual>
        <collision>
            <origin rpy="0 0 0" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://lt_description/meshes/kr10r1100/link_3.STL" />
            </geometry>
        </collision>
    </link>
    <joint name="joint_a3" type="revolute">
        <origin rpy="0 0 0" xyz="0.55896 -0.034133 -0.008" />
        <parent link="link_2" />
        <child link="link_3" />
        <axis xyz="0 0 1" />
        <limit effort="100" lower="-0.8377580409572782" upper="2.9321531433504737" velocity="1" />
    </joint>
    <link name="link_4">
        <inertial>
            <origin rpy="0 0 0" xyz="-5.4213E-05 -0.0016695 0.15386" />
            <mass value="0.01" />
            <!-- value="27.356" /> -->
            <inertia ixx="0.24636" ixy="6.8921E-05" ixz="0.00023212" iyy="0.22365" iyz="0.0075463" izz="0.066562" />
        </inertial>
        <visual>
            <origin rpy="0 0 0" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://lt_description/meshes/kr10r1100/link_4.STL" />
            </geometry>
            <material name="">
                <color rgba="0.8 0.8 0.8 1" />
            </material>
        </visual>
        <collision>
            <origin rpy="0 0 0" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://lt_description/meshes/kr10r1100/link_4.STL" />
            </geometry>
        </collision>
    </link>
    <joint name="joint_a4" type="revolute">
        <origin rpy="-1.5708 0 0" xyz="0.025 0.205 0.0755" />
        <parent link="link_3" />
        <child link="link_4" />
        <axis xyz="0 0 1" />
        <limit effort="100" lower="-5.235987755982989" upper="6.1086523819801535" velocity="1" />
    </joint>
    <link name="link_5">
        <inertial>
            <origin rpy="0 0 0" xyz="0.00037113 -0.053077 0.011452" />
            <mass value="0.01" />
            <!-- value="5.9558" /> -->
            <inertia ixx="0.0097612" ixy="5.2407E-06" ixz="-4.0674E-05" iyy="0.0085564" iyz="7.7038E-05" izz="0.0066757" />
        </inertial>
        <visual>
            <origin rpy="0 0 0" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://lt_description/meshes/kr10r1100/link_5.STL" />
            </geometry>
            <material name="">
                <color rgba="0.8 0.8 0.8 1" />
            </material>
        </visual>
        <collision>
            <origin rpy="0 0 0" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://lt_description/meshes/kr10r1100/link_5.STL" />
            </geometry>
        </collision>
    </link>
    <joint name="joint_a5" type="revolute">
        <origin rpy="0 0 0" xyz="-0.0002289 0.0545 0.31" />
        <parent link="link_4" />
        <child link="link_5" />
        <axis xyz="0.0042 -0.99999 0" />
        <limit effort="100" lower="-1.5707963267948966" upper="1.9198621771937625" velocity="1" />
    </joint>
    <link name="link_6">
        <inertial>
            <origin rpy="0 0 0" xyz="0.00014555 -0.0022055 -0.010033" />
            <mass value="0.01" />
            <!-- value="0.59297" /> -->
            <inertia ixx="0.00022595" ixy="-5.5738E-07" ixz="-2.5386E-06" iyy="0.00023504" iyz="4.1459E-05" izz="0.00041455" />
        </inertial>
        <visual>
            <origin rpy="0 0 0" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://lt_description/meshes/kr10r1100/link_6.STL" />
            </geometry>
            <material name="">
                <color rgba="0.2 0.2 0.2 1" />
            </material>
        </visual>
        <collision>
            <origin rpy="0 0 0" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://lt_description/meshes/kr10r1100/link_6.STL" />
            </geometry>
        </collision>
    </link>
    <joint name="joint_a6" type="revolute">
        <origin rpy="-2.9253 0 0" xyz="0.0011204 -0.054496 0.066994" />
        <parent link="link_5" />
        <child link="link_6" />
        <axis xyz="0.013305 -0.21466 -0.9766" />
        <limit effort="100" lower="-6.1086523819801535" upper="6.1086523819801535" velocity="1" />
    </joint>
    <link name="tool0" />
    <joint name="joint_a6_tool0" type="fixed">
        <!-- <origin
        xyz="0 0 0.33376"
        rpy="0 3.14159 1.5708" /> -->
        <origin rpy="0 0 0" xyz="0 0 0" />
        <parent link="link_6" />
        <child link="tool0" />
    </joint>
    <transmission name="trans_joint_a1">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint_a1">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="joint_a1_motor">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="trans_joint_a2">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint_a2">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="joint_a2_motor">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="trans_joint_a3">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint_a3">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="joint_a3_motor">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="trans_joint_a4">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint_a4">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="joint_a4_motor">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="trans_joint_a5">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint_a5">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="joint_a5_motor">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="trans_joint_a6">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint_a6">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="joint_a6_motor">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <gazebo>
        <plugin name="gazebo_ros_control">
            <robotNamespace>/</robotNamespace>
        </plugin>
    </gazebo>
</robot>

